### PR TITLE
Replace bundled Mono.Options in Converter

### PR DESCRIPTION
**Pull request by Nihlus**
This PR replaces the source-bundled Mono.Options in Generator.Converter with an open package from NuGet instead ([CommandLineParser](https://github.com/gsscoder/commandline)).

This reduces the code footprint of the program, removes a source file which is under a foreign license, and improves the usability of the program.

Furthermore, the readme file has been renamed to conform to naming standards and has had its contents updated.

Summary of changes:

* Replaced Mono.Options with CommandLine
* The options are now automatically mapped to properties in an options class
* Both short and long option syntax is now supported
* The help screen in the program is now automatically generated and up to date with the current code
* The readme has been updated to reflect this
* The target framework of Generator.Converter has been bumped to 4.6.1 in order to support CommandLine
* Paket has updated references in Rewrite and Tests as a side effect.

### CODE REVIEW

File: paket.dependencies [hunk 1]
```diff
@@ -2,11 +2,12 @@ source https://nuget.org/api/v2
...
+nuget CommandLineParser
 nuget FSharp.Formatting
 nuget FsCheck
 nuget FsCheck.Xunit
 nuget xunit.runner.console
 nuget xunit.assert
 nuget FAKE
 nuget Mono.Cecil >= 0.10.0-beta1
-nuget gtk-sharp3
+nuget gtk-sharp3
```
Comment (Frassle) [inline paket.dependencies]: Really annoying how inconsistent final new lines are in this repo :(

File: src/Generator.Rewrite/Generator.Rewrite.csproj [hunk 2]
```diff
@@ -110,7 +110,7 @@
         </Reference>
       </ItemGroup>
     </When>
-    <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And ($(TargetFrameworkVersion) == 'v4.0' Or $(TargetFrameworkVersion) == 'v4.5' Or $(TargetFrameworkVersion) == 'v4.5.1' Or $(TargetFrameworkVersion) == 'v4.5.2' Or $(TargetFrameworkVersion) == 'v4.5.3' Or $(TargetFrameworkVersion) == 'v4.6' Or $(TargetFrameworkVersion) == 'v4.6.1' Or $(TargetFrameworkVersion) == 'v4.6.2' Or $(TargetFrameworkVersion) == 'v4.6.3' Or $(TargetFrameworkVersion) == 'v4.7')">
```
Comment (Frassle) [inline src/Generator.Rewrite/Generator.Rewrite.csproj]: Why is this file changed at all? Isn't this pull request just for Convert?

File: tests/OpenTK.Tests/OpenTK.Tests.fsproj [hunk 3]
```diff
@@ -90,7 +90,7 @@
         </Reference>
       </ItemGroup>
     </When>
-    <When Condition="($(TargetFrameworkIdentifier) == '.NETStandard' And ($(TargetFrameworkVersion) == 'v1.6' Or $(TargetFrameworkVersion) == 'v2.0')) Or ($(TargetFrameworkIdentifier) == '.NETCoreApp' And ($(TargetFrameworkVersion) == 'v1.0' Or $(TargetFrameworkVersion) == 'v1.1' Or $(TargetFrameworkVersion) == 'v2.0'))">
```
Comment (Frassle) [inline tests/OpenTK.Tests/OpenTK.Tests.fsproj]: Same as for Rewrite.csproj, why is this changed at all?

File: src/Generator.Converter/Main.cs [hunk 4]
```diff
@@ -59,48 +59,22 @@ public int GetHashCode(XNode a)
             }
         }
     }
-    
+
     class EntryPoint
     {
+        static Options CLIOptions;
```
Comment (Frassle) [inline src/Generator.Converter/Main.cs]: Can this be moved into Main as a local var?

File: paket.dependencies [hunk 5]
```diff
@@ -9,5 +9,5 @@ nuget FsCheck.Xunit
 nuget xunit.runner.console
 nuget xunit.assert
 nuget FAKE
-nuget Mono.Cecil >= 0.10.0-beta1
```
Comment (Frassle) [inline paket.dependencies]: This looks Bind related. Nothing to do with convert.

Comment (Nihlus) [inline paket.dependencies]: For some reason, the rewriter ceased working when Paket automatically updated the lock file when CommandLineParser was installed. Updating the Mono.Cecil version fixed this.

File: src/Generator.Rewrite/Generator.Rewrite.csproj (hunk 2 above)
Comment (Nihlus) [inline src/Generator.Rewrite/Generator.Rewrite.csproj]: This is a side effect of how Paket runs. It updates all project files, even if only one is specified. Since the lock file is also updated, I wasn't keen on committing just half of the relevant changes.

File: tests/OpenTK.Tests/OpenTK.Tests.fsproj (hunk 3 above)
Comment (Nihlus) [inline tests/OpenTK.Tests/OpenTK.Tests.fsproj]: Same as Rewrite.

File: src/Generator.Converter/Main.cs (hunk 4 above)
Comment (Nihlus) [inline src/Generator.Converter/Main.cs]: Yes. However, the command line arguments are one of the few things that should perhaps be statically accessible to the program - it doesn't change, and it's the same across the entire program no matter the state.

Thoughts?

Comment (Frassle) [inline src/Generator.Converter/Main.cs]: I'm ok with that.

**Comment by Frassle** [comment]
Looks ok to me.
